src/ProtocolZoo/ProtocolZoo.jl

```diff
@@ -305,4 +307,66 @@ end
     end
 end
 
+"""
+$TYPEDEF
+
+A protocol running between two nodes, checking periodically for any entangled pairs between the two nodes and consuming/emptying the qubit slots.
+
+$FIELDS
+"""
+@kwdef struct EntanglementConsumer <:AbstractProtocol
+    """time-and-schedule-tracking instance from `ConcurrentSim`"""
+    sim::Simulation
+    """a network graph of registers"""
+    net::RegisterNet
+    """the vertex index of node A"""
+    nodeA::Int
+    """the vertex index of node B"""
+    nodeB::Int
+    """stores the time and resulting observable from querying nodeA and nodeB for `EntanglementCounterpart`"""
+    log::Vector{Any}
+    """Time period between successive queries on the nodes"""
+    period::Float64
+end
+
+@resumable function (prot::EntanglementConsumer)(;_prot::EntanglementConsumer=prot)
+    prot = _prot # weird workaround for no support for `struct A a::Int end; @resumable function (fa::A) return fa.a end`; see https://github.com/JuliaDynamics/ResumableFunctions.jl/issues/77
+    while true
+        query1 = query(prot.net[prot.nodeA], EntanglementCounterpart, prot.nodeB, ❓)
+        if isnothing(query1)
+            push!(prot.log, (now(prot.sim), nothing, nothing))
+            @yield timeout(prot.sim, prot.period)
+            @debug "EntanglementConsumer between $(prot.nodeA) and $(prot.nodeB): query1 failed"
+            continue
+        else
+            query2 = query(prot.net[prot.nodeB], EntanglementCounterpart, prot.nodeA, query1.slot.idx)
+
+            if isnothing(query2) # in case EntanglementUpdate hasn't reached the second node yet, but the first node has the EntanglementCounterpart
+                push!(prot.log, (now(prot.sim), nothing, nothing))
+                @yield timeout(prot.sim, prot.period)
+                @debug "EntanglementConsumer between $(prot.nodeA) and $(prot.nodeB): query2 failed"
+                continue
+            end
+        end
+        
+        q1 = query1.slot 
+        q2 = query2.slot
+        @debug "EntanglementConsumer between $(prot.nodeA) and $(prot.nodeB): queries successful, consuming entanglement"
+        untag!(q1, query1.tag)
+        untag!(q2, query2.tag)
+        # TODO do we need to add EntanglementHistory and should that be a different EntanglementHistory since the current one is specifically for SwapperProt
+        # TODO currently when calculating the observable we assume that EntanglerProt.pairstate is always (|00⟩ + |11⟩)/√2, make it more general for other states
+        ob1 = observable((q1, q2), Z⊗Z)
+        ob2 = observable((q1, q2), X⊗X)
+
+        prot.net[prot.nodeA].staterefs[q1.idx] = nothing
```

Comment:
We should avoid doing such low-level operations in the protocols. This should be just a traceout operation.

Currently this probably breaks some invariants -- what happens if these two slots were entangled to some third slot -- it will lead to a ton of issues if someone tries to do something with that third slot.